patches/net/minecraft/world/item/ItemStack.java.patch

```diff
@@ -140,6 +140,28 @@
              return list;
          }
      }
+@@ -897,10 +_,21 @@
+         return !this.getOrDefault(DataComponents.ENCHANTMENTS, ItemEnchantments.EMPTY).isEmpty();
+     }
+ 
++    /**
++     * @deprecated Neo: Use {@link #getTagEnchantments()} for NBT enchantments, or {@link #getAllEnchantments} for gameplay.
++     */
++    @Deprecated
+     public ItemEnchantments getEnchantments() {
+         return this.getOrDefault(DataComponents.ENCHANTMENTS, ItemEnchantments.EMPTY);
+     }
+ 
++    /**
++     * Gets all enchantments from NBT. Use {@link ItemStack#getAllEnchantments} for gameplay logic.
++     */
++    public ItemEnchantments getTagEnchantments() {
++        return this.getOrDefault(DataComponents.ENCHANTMENTS, ItemEnchantments.EMPTY);
```

Comment:
Do we want to default this to calling `getEnchantments` rather than duplicating the code inside of it? Also could you move this method above `getEnchantments` so that it is a single chunk of stuff being patched in (even though I am aware that regardless it still is just one hunk)